Various stuff

lipusal opened the pull request:
Various improvements and fixes

- Fall back to largest page size when setting a `page_size` that is too big in Search
- Improve profile picture experience
  - [ ] Solve #154 and fix 
    - Use `profilePictureUrl` field. If `null`, fall back to default URL defined in front.
- Don't get filters in production, a conditional was flipped
- Use bigger pictures in lightbox (Closes #12)
- Etc.
- More to come

## TODO
- [x] Show success SWAL when profile picture is successfully uploaded
- [ ] Handle Twitch client errors

lipusal committed:
Fall back to largest page size if page size is too big

lipusal committed:
Improve profile picture experience

- Remove unnecessary file name field
- Change picture (faded) immediately after selecting
- Show loading circle while uploading
- Use SWAL for upload errors
- Use SWAL to confirm picture deletion

lipusal committed:
Don't get filters in production

lipusal committed:
Improve error message

lipusal commented on paw-webapp/model/src/main/java/ar/edu/itba/paw/webapp/model/validation/ValueErrorConstants.java:
```diff
@@ -46,7 +46,7 @@
 
     public static final ValueError PICTURE_TOO_BIG = new ValueError(ILLEGAL_VALUE, "picture",
-            "The picture is too big");
+            String.format("The picture is too big. Maximum size is %dMB.", NumericConstants.PROFILE_PICTURE_MAX_SIZE / 1000000));
```
@juanmbellini Esto lo cambié porque se lo muestro directamente al usuario. Era bastante choto tener un mensaje que diga "picture is too big" sin decir el tamaño máximo. Ahora además dice "Maximum size is 2MB"

juanmbellini replied:
Te conviene poner ```{0}``` en vez de ```%d```.
A todo esto, por que el dividido 1000000?

juanmbellini replied:
Aaaah el dividido es porque lo pasas de B a MB no?

lipusal replied:
Así es. No sabía que se podía poner `{0}`, creí que esto formateaba a la C. Tanto cambia?

juanmbellini replied:
No se ni siquiera si funciona de la manera que lo pusiste jjaja. Tengo entendido que son parámetros ordenados

lipusal replied:
Sí sí anda, fijate que en [este GIF](https://user-images.githubusercontent.com/5761629/33811738-6a0e49ea-ddf5-11e7-9164-822a0530bce5.gif) aparece el mensaje bien formateado. Yo siempre usé `String.format` así, tampoco sabía que se podía hacer de tu manera así que los dos aprendimos algo ¯\\_(ツ)_/¯

lipusal commented:
## Improved profile picture experience
![a](https://user-images.githubusercontent.com/5761629/33811696-1a326fe6-ddf5-11e7-9632-5741198da8ea.gif)

### When picture is too big
![a](https://user-images.githubusercontent.com/5761629/33811738-6a0e49ea-ddf5-11e7-9164-822a0530bce5.gif)

Still have to stop showing "delete profile picture" all the time

lipusal commented:
Les parece que debería mostrar un SWAL (el modal con el tick o la cruz cuando hay error) cuando la foto se sube bien? Me pareció medio al pedo porque la foto pasa de tener 50% de opacidad a 100%, y que con eso alcanza como para "confirmar" que la foto está subida. Además hay un loading circle que desaparece cuando termina, y el submit button cambia de `Uploading...` a `Submit` de nuevo y se rehabilita.

lipusal committed:
Profile picture improvements

- Fade out picture correctly
- Change button text

juanmbellini commented:
@lipusal No se de qué hablas con eso de SWAL hjajaja

lipusal commented:
@juanmbellini SweetAlert, el modal que salta en las fotos que subí (`Delete profile picture?`, `Picture deleted`, etc.). Estoy preguntando si poner uno de success cuando la foto se sube bien, para mí no hace falta porque hay suficientes indicaciones en otros lugares (ver mi comentario anterior), pero quizá esté bueno explicitarlo.

juanmbellini commented:
@lipusal mmmm para mí esta bueno darle todo tipo de feedback al usuario. Mejora la UX

lipusal committed:
Add "Unknown" for empty game properties

lipusal commented:
### Add "Unknown" for empty game properties

### Before
![image](https://user-images.githubusercontent.com/5761629/33812414-056ae6d6-ddfc-11e7-88b3-3a0422a69b94.png)
----
### After
![image](https://user-images.githubusercontent.com/5761629/33812409-fbd69778-ddfb-11e7-9a4e-5d15a0174515.png)

lipusal committed:
Bring back lazy loading

lipusal committed:
Use bigger game screenshots

lipusal commented:
## Use bigger pictures in lightbox

### Before
![before](https://user-images.githubusercontent.com/5761629/33813497-d66a539c-de02-11e7-9a02-0db99fd1ff41.gif)

### After
![after](https://user-images.githubusercontent.com/5761629/33813501-ddd142e4-de02-11e7-8b70-93e500627965.gif)

lipusal committed:
Fix some attributes

lipusal committed:
Open game in new tab in Write Review page

lipusal committed:
Fix NPE in UserServiceImpl#getWithSocialStuff

lipusal commented:
@juanmbellini El NPE lo arreglé en 40b2dbffb36512aec8439fa1254c0794c49f0a69, no me salía con optionals y era un chequeo boludo así que lo dejé así, sino tenía que cambiar todas las demás funciones que usaban al método

juanmbellini commented:
@lipusal #154 ya tiene PR (#159)

Drocheg committed:
Stuff profile (#157)

* Fix create shelfs by changing ng-if to ng-show that was messing up a variable.
Adds the list of shelf to the profile page.

* Fix updating games by changing ng-if to ng-show that was messing up a variable.

* -Changed writing reviews to have the score and the statuses and shelves.
-fixes problem in games where shelves where badly loaded because of race condition
-added in shelves button to delete game and button to edit
-added links in profile page to gameList

* -deleted materialize thing from build.js

* -deleted unused code commented.

* -undeleted autocomplete="off" that was there from before

* -Deleted extra line in write-review html that was added to test.

* -Fixes minor errors

lipusal committed:
Merge branch 'master' into stuff

# Conflicts:
#	app/views/home.html

lipusal committed:
Merge remote-tracking branch 'origin/stuff' into stuff

# Conflicts:
#	app/scripts/controllers/HomeCtrl.js

juanmbellini commented:
@lipusal #154 ya fue cerrado con #159 que fue mergeada. Te marque la task

juanmbellini commented:
@lipusal #159 ya fue mergeado. Te marque la task

Drocheg reviewed (approved):
buenisimo +10
